src/test/unit_tests/test_namespace_fs.js

```diff
@@ -446,6 +446,116 @@ mocha.describe('namespace_fs', function() {
+                key: restore_key,
+                source_stream: buffer_utils.buffer_to_read_stream(data)
+            }, dummy_object_sdk);
+
+            console.log('upload_object response', inspect(upload_res));
+        });
+
+        mocha.it('fails when GLACIER storage class is not supported', async function() {
+            try {
+                await ns_tmp.restore_object({
+                    bucket: upload_bkt,
+                    key: restore_key,
+                }, dummy_object_sdk);
+
+                assert.fail('restore_object should fail');
+            } catch (err) {
+                assert.strictEqual(err.rpc_code, 'NOT_IMPLEMENTED');
+            }
+
+            mocha.it('restore_object with GLACIER supported', function() {
+                const temp_restore_key = restore_key + "_glacier";
+
+                mocha.before(async function() {
+                    // Monkey patch the _is_storage_class_supported function to return true
+                    ns_tmp._is_storage_class_supported = async () => true;
+
+                    console.log("Uploading object to GLACIER storage class");
+                    const upload_res_2 = await ns_tmp.upload_object({
+                        bucket: upload_bkt,
+                        key: temp_restore_key,
+                        source_stream: buffer_utils.buffer_to_read_stream(data),
+                        storage_class: s3_utils.STORAGE_CLASS_GLACIER,
+                    }, dummy_object_sdk);
+
+                    console.log('upload_object response 2', inspect(upload_res_2));
+                });
+
+                mocha.it('fails when the uploaded object is not is GLACIER storage class', async function() {
+                    try {
+                        await ns_tmp.restore_object({
+                            bucket: upload_bkt,
+                            key: restore_key,
+                        }, dummy_object_sdk);
+
+                        assert.fail('restore_object should fail');
+                    } catch (err) {
+                        assert.strictEqual(err.rpc_code, 'INVALID_OBJECT_STATE');
+                    }
+                });
+
+                mocha.it('succeeds when object is in GLACIER storage class', async function() {
+                    const restore_res = await ns_tmp.restore_object({
+                        bucket: upload_bkt,
+                        key: temp_restore_key,
+                    }, dummy_object_sdk);
+
+                    console.log('restore_object response', inspect(restore_res));
+
+                    const restore_objectmd = await ns_tmp.read_object_md({
+                        bucket: upload_bkt,
+                        key: temp_restore_key,
+                    }, dummy_object_sdk);
+
+                    assert.strictEqual(restore_objectmd.storage_class, s3_utils.STORAGE_CLASS_GLACIER);
+                    assert.strictEqual(restore_objectmd.xattr['user.noobaa.restore.request'], "false");
```

Comment:
read_object_md returns user.noobaa xattrs? we should extract internal xattrs from read_object_md response

Reply:
@romayalon, thank you for catching this. The test was not even running for some odd reason (probably because if nested `it()`, I don't know).

I have fixed the test now.

Reply:
so now which xattrs do you see in the test?

Reply:
I am no longer relying on getting xattr from read_object_md. Rather I use a helper function `get_xattr` (or something like that),

Reply:
yes but do we still have the xattr in read_object_md reply? because we shouldn't

Reply:
No we don't have these xattrs in the reply. I used them accidentally in the test and the tests passed because they apparently weren't even running.

Just to be sure, as you can, I have just added assertions in the tests as well now which ensures that the xattrs are indeed `undefined`.

Reply:
perfect thanks